decompiler/analysis/final_output.cpp

```diff
@@ -218,7 +218,25 @@ std::string careful_function_to_string(
   }
 
   if (!func->ir2.top_form) {
-    return ";; ERROR: function was not converted to expressions. Cannot decompile.\n\n";
+    if (func->mips2c_output) {
+      std::string output = ";; INFO: function output is handled by mips2c\n";
+      // Attempt to automatically generate the OpenGOAL code for calling the mips2c varient
+      // For methods this is - (defmethod-mips2c "(method <METHOD_ID> <TYPE>)" <METHOD_ID>)
+      if (func->guessed_name.kind == FunctionName::FunctionKind::METHOD) {
+        output += fmt::format("(defmethod-mips2c \"(method {} {})\" {} {})\n",
+                              func->guessed_name.method_id, func->guessed_name.type_name,
+                              func->guessed_name.method_id, func->guessed_name.type_name);
+      } else if (func->guessed_name.kind == FunctionName::FunctionKind::GLOBAL) {
+        // For functions it is - (def-mips2c <FUNC_NAME> (function <ARGS> <RET>))
+        // TODO - is it possible to easily get the args from `all-types` to this point?
+        output += fmt::format(";; (def-mips2c {} (function ARGS RET))\n",
+                              func->guessed_name.function_name);
```

Comment:
The full type should be `func->type.print()`. The name stuff is just the name used to identify the function and doesn't have the full type information.